store/multi_writer_app_store.go

```diff
@@ -167,12 +206,30 @@ func (s *MultiWriterAppStore) SaveVersion() ([]byte, int64, error) {
 		// Tie up Patricia tree with IAVL tree.
 		// Do this after the feature flag is enabled so that we can detect
 		// inconsistency in evm.db across the cluster
-		// AppStore 3.1 write EVM root to app.db only if it changes
+		// AppStore 3.1 writes EVM root to app.db only if it changes
 		if bytes.Equal(s.appStore.Get(appStoreVersion3_1), []byte{1}) {
 			oldRoot := s.appStore.Get(rootKey)
 			if !bytes.Equal(oldRoot, currentRoot) {
 				s.appStore.Set(rootKey, currentRoot)
 			}
+
+			// vm keys deletion process
```

Comment:
Why is this nested in here? It's independent of the v3.1 / evm root stuff.

Reply:
move it out